tools/configuration.py

```diff
@@ -0,0 +1,117 @@
+import os
+import sys
+import getopt
+
+from tools import get_pipe_output
+
+default_conf = {
+    'max_domains': 10,
+    'max_ext_length': 10,
+    'style': 'gitstats.css',
+    'max_authors': 7,
+    'max_authors_of_months': 6,
+    'authors_top': 5,
+    'commit_begin': '',
+    'commit_end': 'HEAD',
+    'linear_linestats': 1,
+    'project_name': '',
+    'processes': 8,
+    'start_date': '',
+    'output': 'html'
+}
+
+
+class ConfigurationException(Exception):
```

Comment:
Do not think custom exceptions are useful here. Normally python has enough exception types for such a small script